pkg/platform/provider/baremetal/cluster/update.go

```diff
@@ -140,6 +146,57 @@ func (p *Provider) EnsurePreClusterUpgradeHook(ctx context.Context, c *v1.Cluste
 	return util.ExcuteCustomizedHook(ctx, c, platformv1.HookPreClusterUpgrade, c.Spec.Machines[:1])
 }
 
+func (p *Provider) EnsureUpgradeCoreDNS(ctx context.Context, c *v1.Cluster) error {
+	versionRet, err := version.Parse(c.Spec.Version)
+	if err != nil {
+		return errors.Wrap(err, "unable to update coreDNS version")
+	}
+	if len(versionRet) < 2 {
+		return errors.Wrap(fmt.Errorf("invaild version format %s", c.Spec.Version), "unable to update coreDNS version")
+	}
+	if versionRet[0] == "1" && versionRet[1] == "19" {
+
+		client, err := c.Clientset()
+		if err != nil {
+			return errors.Wrap(err, "unable to update coreDNS version")
+		}
+		err = updateCoreDNSVersion(ctx, client, images.Get().CoreDNS.Tag)
+		if err != nil {
+			return errors.Wrap(err, "unable to update coreDNS version")
+		}
+	} else {
+		logger := log.FromContext(ctx).WithName("Upgrade coreDNS")
+		logger.Infof("Target k8s version is %s, skip upgrade coreDNS", c.Spec.Version)
+	}
+	return nil
+}
+
+func updateCoreDNSVersion(ctx context.Context, client kubernetes.Interface, version string) error {
```

Comment:
please try `kubeadm config upload from-file --config /path/kubeadmconf.yml` to update cm rather than do it manually

Reply:
you should use  `kubeadm config view `  > then `kubeadm config upload from-file --config /path/kubeadmconf.yml`